cranelift/codegen/src/ir/dfg.rs

```diff
@@ -383,9 +379,14 @@ pub enum ValueDef {
 impl ValueDef {
     /// Unwrap the instruction where the value was defined, or panic.
     pub fn unwrap_inst(&self) -> Inst {
+        self.inst().expect("Value is not an instruction result")
+    }
+
+    /// Get the instruction where the value was defined, if any.
+    pub fn inst(&self) -> Option<Inst> {
```

Comment:
You can already match on `ValueDef` yourself.

Reply:
This is true, but it is nice having a convenience method, same as why `Result<T, E>` has an `ok` method.